### PR TITLE
Update eng/common folder

### PR DESCRIPTION
**Pull request by kythant**
This is a manual step now but in the future it should be automatic.
 
Improvements/bug fixes for eng/common folder
- Maestro update wasn't working for AzureDevOps repositories, it works now
- Only trigger subscriptions whose source repository is the same as ours

### CODE REVIEW

File: eng/common/Maestro-PublishBuildToMaestro.yml [hunk 1]
```diff
@@ -103,15 +93,27 @@ steps:
...
       arguments: -url '$(MaestroUri)' -Token '$(MaestroToken)' -api '/api/builds' -jsonBodyPath '$(Build.SourcesDirectory)\eng\common\maestro-build.json'
  
-  - ${{ if eq(parameters.TriggerSubscrption, 'true') }}:
+  - ${{ if eq(parameters.TriggerSubscription, 'true') }}:
     - task: powershell@2
       displayName: 'Publish Build to Default Channel'
       inputs:
         targetType: 'inline'
         script: |
+          . .\eng\common\MaestroHelpers.ps1
```
Comment (huichen123) [inline eng/common/Maestro-PublishBuildToMaestro.yml:102]: >.  [](http://example.com/codeflow?start=10&length=2)

what is this?

File: eng/common/MaestroHelpers.ps1 [hunk 2]
```diff
@@ -6,4 +6,25 @@ function ExtractOrgFromAzureDevOpsCollectionUri([string]$CollectionUri)
...
+    $repository = "https://dev.azure.com/" + $devOpsAccount + "/" + $buildRepositoryUriSplit[3] + "/" + $buildRepositoryUriSplit[4] + "/" + $buildRepositoryUriSplit[5]
+    return $repository
+}
+
+function IsGitHubRepo([string]$buildRepositoryUri)
+{
+    $githubUri = "https://github.com"
+    if ($buildRepositoryUri.length -ge $githubUri.length)
+    {
+        if($buildRepositoryUri.Substring(0, $githubUri.length) -eq $githubUri)
```
Comment (huichen123) [inline eng/common/MaestroHelpers.ps1:24]: >$buildRepositoryUri [](http://example.com/codeflow?start=11&length=19)

is it possible the url is: "[https://www.github.com](https://www.github.com)" or "[http://github.com](http://github.com)"?

File: eng/common/Maestro-PublishBuildToMaestro.yml (hunk 1 above)
Comment (kythant) [reply eng/common/Maestro-PublishBuildToMaestro.yml:102]: https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_scopes?view=powershell-5.1#using-dot-source-notation-with-scope

File: eng/common/MaestroHelpers.ps1 (hunk 2 above)
Comment (kythant) [reply eng/common/MaestroHelpers.ps1:24]: If it is, there is a security risk and we probably want to fail? 
It should always be https

**Review by huichen123 — APPROVED** [review]
:shipit:

**Pull request merged**